girder/utility/s3_assetstore_adapter.py

```diff
@@ -375,6 +375,7 @@ def finalizeUpload(self, upload, file):
                     'url': url,
                     'headers': headers
                 }
+                file['additionalFinalizeKeys'] = ('s3FinalizeRequest',)
```

Comment:
Our API prior to this change was not filtering file models that were getting returned, which was bad. I fixed that on this branch so that file models returned via REST only contain the keys they're supposed to. However, in the S3 case the client needs to know this additional information, so I added this behavior of including additional keys for the finalize step. It's rather specialized, so if you can think of a more elegant way I'm all ears.